src/components/computing-status/use-computing-status.ts

```diff
@@ -76,43 +64,88 @@ function isWorthUpdate(
+    [ComputingType.SHORT_CIRCUIT]: genetateStatusFromComputingType(
+        'shortCircuitAnalysis'
+    ),
+    [ComputingType.SHORT_CIRCUIT_ONE_BUS]: genetateStatusFromComputingType(
+        'oneBusShortCircuitAnalysis'
+    ),
+    [ComputingType.DYNAMIC_SIMULATION]:
+        genetateStatusFromComputingType('dynamicSimulation'),
+    [ComputingType.VOLTAGE_INITIALIZATION]:
+        genetateStatusFromComputingType('voltageInit'),
+    [ComputingType.STATE_ESTIMATION]:
+        genetateStatusFromComputingType('stateEstimation'),
+};
+
+interface UseComputingStatusProps {
+    (
+        fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>,
+        resultConversion: (x: string) => RunningStatus,
+        computingType: ComputingType,
+        optionalServiceAvailabilityStatus?: OptionalServicesStatus
+    ): void;
+}
 /**
  *  this hook loads <computingType> state into redux, then keeps it updated according to notifications
- * @param studyUuid current study uuid
- * @param nodeUuid current node uuid
  * @param fetcher method fetching current <computingType> state
  * @param invalidations when receiving notifications, if updateType is included in <invalidations>, this hook will update
  * @param resultConversion converts <fetcher> result to RunningStatus
  * @param computingType ComputingType targeted by this hook
  * @param optionalServiceAvailabilityStatus status of an optional service
  */
 export const useComputingStatus: UseComputingStatusProps = (
-    studyUuid,
-    nodeUuid,
     fetcher,
-    invalidations,
-    completions,
     resultConversion,
     computingType,
     optionalServiceAvailabilityStatus = OptionalServicesStatus.Up
 ) => {
+    const nodeUuid = useSelector(
+        (state: ReduxState) => state.currentTreeNode?.id
+    );
+    const studyUuid = decodeURIComponent(useParams().studyUuid ?? '') as UUID;
     const nodeUuidRef = useRef<UUID | null>(null);
     const studyUpdatedForce = useSelector(
         (state: ReduxState) => state.studyUpdated
     );
     const lastUpdateRef = useRef<LastUpdateProps | null>(null);
     const dispatch = useDispatch();
+    const status = computingTypeToStatusMapper[computingType];
 
     //the callback crosschecks the computation status and the content of the last update reference
     //in order to determine which computation just ended
     const isComputationCompleted = useCallback(
-        (status: RunningStatus) =>
-            [RunningStatus.FAILED, RunningStatus.SUCCEED].includes(status) &&
-            completions.includes(
+        (runningStatus: RunningStatus) =>
+            [RunningStatus.FAILED, RunningStatus.SUCCEED].includes(
+                runningStatus
+            ) &&
+            status.completions.includes(
                 lastUpdateRef.current?.studyUpdatedForce.eventData?.headers
                     ?.updateType ?? ''
             ),
-        [completions]
+        [status.completions]
```

Comment:
since status isn't memoized, I think this callback will be recreated every time anyway
memoize status or remove this useCallback ?

Reply:
status is a constant because it is defined outside the function and thus becomes a pointer to the fixed memory object